### PR TITLE
adding code sample for Google Cloud Memorystore

### PR DESCRIPTION
**Pull request by jabubake**

### CODE REVIEW

File: memorystore/redis/gce_deployment/startup-script.sh [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+#  limitations under the License.
+# [START startup_script_sh]
+set -ex
+
+# Talk to the metadata server to get the project id and location of application binary.
+PROJECTID=$(curl -s "http://metadata.google.internal/computeMetadata/v1/project/project-id" -H "Metadata-Flavor: Google")
+GCS_APP_LOCATION=$(curl -s "http://metadata.google.internal/computeMetadata/v1/instance/attributes/app-location" -H "Metadata-Flavor: Google")
+WAR=$(curl -s "http://metadata.google.internal/computeMetadata/v1/instance/attributes/app-war" -H "Metadata-Flavor: Google")
+
+gsutil cp "$GCS_APP_LOCATION"** .
```
Comment (lesv) [inline memorystore/redis/gce_deployment/startup-script.sh]: I wonder a bit about this being correct.

File: memorystore/redis/gce_deployment/startup-script.sh [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+# Install dependencies from apt
+apt-get update
+apt install -t jessie-backports -yq openjdk-8-jre-headless ca-certificates-java
```
Comment (lesv) [inline memorystore/redis/gce_deployment/startup-script.sh]: any reason we aren't on Stretch ?

Comment (lesv) [reply memorystore/redis/gce_deployment/startup-script.sh]: ie. debian-9

File: memorystore/redis/gce_deployment/startup-script.sh [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+# Make Java8 the default
+update-alternatives --set java /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java
+
+# Jetty Setup
+mkdir -p /opt/jetty/temp
+mkdir -p /var/log/jetty
+
+# Get Jetty
+curl -L https://repo1.maven.org/maven2/org/eclipse/jetty/jetty-distribution/9.4.4.v20170414/jetty-distribution-9.4.4.v20170414.tar.gz -o jetty9.tgz
```
Comment (lesv) [inline memorystore/redis/gce_deployment/startup-script.sh]: 9.4.10.v20180503 is latest.

File: memorystore/redis/pom.xml [hunk 4]
```diff
@@ -0,0 +1,78 @@
+<!--
+  Copyright 2018 Google Inc.
```
[the remark this answers is not in the document]
Comment (jabubake) [reply memorystore/redis/pom.xml]: Will need to release an update and then these repos (will follow up w/ that cleanup) after merge.

File: memorystore/redis/gke_deployment/visit-counter.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: visit-counter
+    spec:
+      containers:
+      - name: visit-counter
+        image: "gcr.io/<PROJECT_ID>/visit-counter:v1"
```
[the remark this answers is not in the document]
Comment (jabubake) [reply memorystore/redis/gke_deployment/visit-counter.yaml]: Nope :( I do wish there was.

File: memorystore/redis/pom.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (kurtisvg) [reply memorystore/redis/pom.xml]: We released the update for java-repo-tools earlier today (v1.0.10)

[the remark this answers is not in the document]
Comment (jabubake) [reply memorystore/redis/pom.xml]: Woot :) great, updated.

File: memorystore/redis/src/main/java/com/example/redis/AppServletContextListener.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2018 Google Inc.
```
[the remark this answers is not in the document]
Comment (jabubake) [reply memorystore/redis/src/main/java/com/example/redis/AppServletContextListener.java]: done

File: memorystore/redis/src/main/java/com/example/redis/VisitCounterServlet.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2018 Google Inc.
```
[the remark this answers is not in the document]
Comment (jabubake) [reply memorystore/redis/src/main/java/com/example/redis/VisitCounterServlet.java]: done

File: memorystore/redis/gke_deployment/visit-counter.yaml [hunk 8]
```diff
@@ -0,0 +1,39 @@
+
```
[the remark this answers is not in the document]
Comment (jabubake) [reply memorystore/redis/gke_deployment/visit-counter.yaml]: done

File: memorystore/redis/src/main/resources/application.properties [hunk 9]
```diff
@@ -0,0 +1,16 @@
+#  Copyright 2018 Google Inc.
```
[the remark this answers is not in the document]
Comment (jabubake) [reply memorystore/redis/src/main/resources/application.properties]: done

**Comment by jabubake** [comment]
@lesv @kurtisvg PTAL

**Pull request merged**